src/core_tests.rs

```diff
@@ -178,6 +187,27 @@ impl TestNode {
     }
 }
 
+// Prints all nodes in a table, organised by group.
+// Does not check the routing tables (use verify_invariant_for_all_nodes() for that).
+fn print_nodes(nodes: &[TestNode], clients: &[TestClient]) {
+    let mut groups = HashMap::<Prefix<XorName>, HashSet<XorName>>::new();
+    for node in nodes {
+        groups.entry(node.group_prefix())
+            .or_insert_with(HashSet::new)
+            .insert(node.name());
+    }
+    for (prefix, names) in groups {
+        println!("Group {:?}:", prefix);
+        for name in names {
+            println!("Node\t{:?}", name);
+        }
+    }
+    println!("Clients:");
+    for client in clients {
+        println!("\t{:?}", client.name());
```

Comment:
Agree on both points.  Using `trace!` rather than `println!` means the statements yield more info, can be filtered out without recompiling, and can be handled properly by editors which are configured to handle logging syntax.